crates/pink/sidevm/env/src/tasks.rs

```diff
@@ -18,7 +18,12 @@ thread_local! {
     /// The id of the current polling task. Would be passed to each ocall.
     static CURRENT_TASK: std::cell::Cell<i32>  = Default::default();
     /// All async tasks in the sidevm guest.
-    static TASKS: RefCell<Tasks> = RefCell::new(vec![Some(unsafe { sidevm_main_future() })]);
+    static TASKS: RefCell<Tasks> = {
+        std::panic::set_hook(Box::new(|info| {
```

Comment:
Does it run when initializing the static variable? Not sure why it's set here.

Reply:
Yes. Set it here because this is the place where we create the main future (invoke the `async fn main()`). So we can think of it as the premain code for a normal program.